### DISCUSSION TITLE
[Files Service] Configuring multiple buckets for the same instance

### DISCUSSION THREAD

**Discussion by seraph-eeno** [mia-platform/community#84] [Q&A] [answered]
Hello guys,
in my team we are currently in the situation in which we have three buckets AWS which host different types of utility files. Therefore, we have three microservices implementing the file service, each with its own specific bucket configuration. So, having three AWS buckets, we built up three different microservices for the file service.

At this point, i wouldn't want to have to set up a new file service each time a new bucket is needed, mostly because of resources optimization in the cluster. From the [doc](https://docs.mia-platform.eu/docs/runtime_suite/files-service/configuration#configuration-file) it seems that it's possible to configure only one bucket per cloud technology for each file service instance.

Is it possible to describe an array of different buckets in the configurations for the same file service instance? E.g., three AWS buckets, two Google Cloud and so on...
If not, is there any way for this to be added as a feature in the file service?

**Comment by fredmaggiowski** [comment, marked as answer]
Hi, as you correctly noted there's currently no way to setup a files service instance that interacts with multiple buckets; however this is not the first time that this request has been done.

I guess this could be a feature that could be supported by the service; if you want you can try and open a PR on the service repository with your proposal (starting from the documentation on which we could discuss before starting with the implementation).

**Reply by seraph-eeno** [reply]
Hi Federico, as suggested I've opened a PR [here](https://git.tools.mia-platform.eu/platform/plugins/files-service/-/merge_requests/136).
